autosklearn/data/validation.py

```diff
@@ -26,603 +21,115 @@ class InputValidator:
-            return y
-        else:
-
-            return self.target_encoder.transform(y)
-
-    def decode_target(
+    ) -> BaseEstimator:
+        """
+        Validates and fit a categorical encoder (if needed) to the features, and
+        a encoder for targets in the case of classification.
+
+        Dimensionality and data types checks are also performed
+
+        Parameters
+        ----------
+            X_train: typing.Union[np.ndarray, pd.DataFrame]
+                A set of features that are going to be validated (type and dimensionality
+                checks) and a encoder fitted in the case the data needs encoding
+            y_train: typing.Union[np.ndarray, pd.DataFrame]
+                A set of targets that are going to be encoded for classification
+            X_test: typing.Union[np.ndarray, pd.DataFrame]
+                A hold out set of features used for checking
+            y_test: typing.Union[np.ndarray, pd.DataFrame]
+                A hold out set of targets used for checking
+            feat_type: typing.Optional[typing.List[str]]
+                In case the data is not a pandas DataFrame, this list indicates
+                which columns should be treated as categorical
+            is_classification: bool
+                For classification task, this flag indicates that the target data
+                should be encoded
+        """
+        # Check that the data is valid
+        self.check_data(X_train, y_train, X_test, y_test)
+        self.feature_validator.fit(X_train, X_test, feat_type)
+        self.target_validator.fit(y_train, y_test, is_classification)
+        self._is_fitted = True
+
+        return self
+
+    def transform(
         self,
-        y: np.ndarray,
+        X: typing.Union[np.ndarray, pd.DataFrame],
+        y: typing.Optional[typing.Union[np.ndarray, pd.DataFrame]] = None,
     ) -> np.ndarray:
         """
-        If the original target features were encoded,
-        this method employs the inverse transform method of the encoder
-        to decode the original features
-        """
-        if self.target_encoder is None:
-            return y
-
-        # Handle different ndim encoder for target
-        if hasattr(self.target_encoder, 'inverse_transform'):
-            return self.target_encoder.inverse_transform(y)
+        Transform the given target or features to a numpy array
+
+        Parameters
+        ----------
+            X: typing.Union[np.ndarray, pd.DataFrame]
+                A set of features to transform
+            y: typing.Union[np.ndarray, pd.DataFrame]
+                A set of targets to transform
+
+        Return
+        ------
+            np.ndarray:
+                The transformed features array
+            np.ndarray:
+                The transformed targets array
+        """
+        if not self._is_fitted:
+            raise ValueError("Cannot call transform on a validator that is not fitted")
```

Comment:
Could you please raise https://scikit-learn.org/stable/modules/generated/sklearn.exceptions.NotFittedError.html instead?

Reply:
Done!